corporate/lib/stripe_event_handler.py

```diff
@@ -150,3 +152,35 @@ def handle_payment_intent_succeeded_event(
         False,
         billing_session.get_remote_server_legacy_plan(payment_intent.customer),
     )
+
+
+@error_handler
+def handle_invoice_payment_succeeded(stripe_invoice: stripe.Invoice, invoice: Invoice) -> None:
+    invoice.status = Invoice.PAYMENT_SUCCEEDED
+    invoice.save(update_fields=["status"])
+
+    customer = invoice.customer
+
+    configured_fixed_price_plan = None
+    if customer.required_plan_tier:
+        configured_fixed_price_plan = get_configured_fixed_price_plan_offer(
+            customer, customer.required_plan_tier
+        )
+
+    if stripe_invoice.collection_method == "send_invoice" and configured_fixed_price_plan:
+        billing_session = get_billing_session_for_stripe_webhook(customer, user_id=None)
+        remote_server_legacy_plan = billing_session.get_remote_server_legacy_plan(customer)
+        assert customer.required_plan_tier is not None
+        billing_session.process_initial_upgrade(
+            plan_tier=customer.required_plan_tier,
+            # TODO: Currently licenses don't play any role for fixed price plan.
+            # We plan to introduce max_licenses allowed soon.
+            licenses=0,
+            automanage_licenses=True,
+            billing_schedule=CustomerPlan.BILLING_SCHEDULE_ANNUAL,
+            charge_automatically=False,
+            free_trial=False,
+            remote_server_legacy_plan=remote_server_legacy_plan,
+            # TODO: Should we decide for whether to schedule or not via metadata or support ?
+            should_schedule_upgrade_for_legacy_remote_server=remote_server_legacy_plan is not None,
```

Comment:
Since the customer has already paid, this is a direct upgrade.

Reply:
For scheduled upgrades, we wait till the days of the upgrade to charge customer. So, scheduling an upgrade will cause problems anyway.

Reply:
Ah, that's right! I'll update.